SimPEG/utils/code_utils.py

```diff
@@ -1070,7 +1070,7 @@ def validate_type(property_name, obj, obj_type, cast=True, strict=False):
     if cast:
         try:
             obj = obj_type(obj)
-        except:
+        except TypeError:
```

Comment:
For this one, there are just so many possible error types that could happen, I imagine that the best way to do this one would be something like:

```
except Exception as err:
    raise TypeError(
                f"{type(obj).__name__} cannot be converted to type {obj_type.__name__} "
                f"required for {property_name}."
            ) from err
```

Reply:
Done!